lib/buffer.js

```diff
@@ -807,8 +807,8 @@ Buffer.prototype.toJSON = function() {
 
 
 function adjustOffset(offset, length) {
-  offset |= 0;
-  if (offset === 0) {
+  offset = Math.floor(+offset);
+  if (offset === 0 || Number.isNaN(offset)) {
```

Comment:
Since `slice()` supports negative numbers `Math.round()` must be used. A simple test for that would be
```js
const buf = Buffer.from('abc');
assert.strictEqual(buf.slice(-0.5).toString(), buf.toString();
```